Source/WebKit/WPEPlatform/wpe/WPERectangle.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2024 Igalia S.L.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+ * PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "WPERectangle.h"
+
+#include <wtf/FastMalloc.h>
+
+/**
+ * WPERectangle:
+ * @x: The X coordinate of the top-left corner of the rectangle.
+ * @y: The Y coordinate of the top-left corner of the rectangle.
+ * @width: The width of the rectangle.
+ * @height: The height of the rectangle.
+ *
+ * Boxed type representing a rectangle with integer coordiantes.
+ */
+
+static WPERectangle* wpe_rectangle_copy(WPERectangle* rectangle)
+{
+    g_return_val_if_fail(rectangle, nullptr);
+
+    WPERectangle* copy = static_cast<WPERectangle*>(fastZeroedMalloc(sizeof(WPERectangle)));
+    *copy = *rectangle;
+    return copy;
```

Comment:
Instead of allocating zeroed memory and then immediately overwriting its contents, we can use `fastMemDup()` instead:

```suggestion
    return static_cast<WPERectangle*>(fastMemDup(rectangle, sizeof(WPERectangle)));
```